app/lib/features/invite_members/widgets/invite_code_ui.dart

```diff
@@ -0,0 +1,142 @@
+import 'dart:math';
+
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/toolkit/buttons/inline_text_button.dart';
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/dotted_border_widget.dart';
+import 'package:acter/features/super_invites/providers/super_invites_providers.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::invite_members::invite_code');
+
+class InviteCodeUI extends ConsumerStatefulWidget {
+  final String roomId;
+  const InviteCodeUI({super.key, required this.roomId});
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _InviteCodeUIState();
+}
+
+class _InviteCodeUIState extends ConsumerState<InviteCodeUI> {
+  @override
+  Widget build(BuildContext context) {
+    var inviteCode = ref
+        .watch(inviteCodeForSelectRoomOnly(widget.roomId))
+        .valueOrNull
+        ?.token();
+    if (inviteCode != null) {
+      return Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          DottedBorderWidget(
+            child: Row(
+              children: [
+                Expanded(
+                  child: Text(
+                    inviteCode,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+                IconButton(
+                  onPressed: () {
+                    Clipboard.setData(
+                      ClipboardData(text: inviteCode),
+                    );
+                  },
+                  icon: const Icon(Icons.copy),
+                ),
+              ],
+            ),
+          ),
+          const SizedBox(height: 10),
+          Align(
+            alignment: Alignment.centerRight,
+            child: ActerInlineTextButton(
+              onPressed: () async {
+                final token =
+                    await ref.read(superInviteTokenProvider(inviteCode).future);
+                if (!context.mounted) return;
```

Comment:
Not quit understand the usage of 
`final token = await ref.read(superInviteTokenProvider(inviteCode).future);`

Can I have more information about this?

Reply:
needed for routing. the target expect `extra: SuperInvitesToken` at the moment and I didn't want to refactor routing just for that at the moment (though it is bad routing). Instead I fetch the token (we already have anyways) from the provider via its code and pass it along.

Reply:
`inviteCode` isn't work with route param?